numpy/linalg/_linalg.py

```diff
@@ -3679,3 +3684,264 @@ def vecdot(x1, x2, /, *, axis=-1):
 
     """
     return _core_vecdot(x1, x2, axis=axis)
+
+# Generalized eigenvalue problem
+def _geneigvals_dispatcher(A, B):
+    return (A, B)
+
+@array_function_dispatch(_geneigvals_dispatcher)
+def geneigvals(A, B):
+    """
+    Solve the generalized eigenvalue problem Av = λBv.
+
+    Parameters
+    ----------
+    A :  matrix.
+    B :  matrix.
+
+    Returns
+    -------
+    eigenvalues :
+        The generalized eigenvalues λ.
+    eigenvectors :
+        The generalized eigenvectors v, such that Av = λBv.
+
+    Notes
+    -----
+    This function solves the problem Av = λBv, i.e., finds λ and v such that
+    det(A - λB) = 0. If B is invertible, this is equivalent to the standard
+    eigenvalue problem for B^{-1}A, but this function does not require B to be
+    invertible and does not explicitly invert B.
+    """
+    # Convert to arrays and check shapes
+    A, _ = _makearray(A)
+    B, wrap = _makearray(B)
+    _assert_stacked_square(A, B)
+    _assert_finite(A, B)
+    t, result_t = _commonType(A, B)
+
+    # Try to reduce to standard eigenvalue problem if B is invertible
+    try:
+        Binv = inv(B)
+        BinvA = matmul(Binv, A)
+        result = eigvals(BinvA)
+        return result
+    except LinAlgError:
+        raise LinAlgError(
+            "B must be invertible"
+        )
+
+def _geneig_dispatcher(A, B):
+    return (A, B)
+
+@array_function_dispatch(_geneig_dispatcher)
+def geneig(A, B):
```

Comment:
same question as above here

```suggestion
def _geneig_dispatcher(A, B, /):
    return A, B

@array_function_dispatch(_geneig_dispatcher)
def geneig(A, B, /):
```